be/src/object_store/s3_object_store.cpp

```diff
@@ -16,53 +15,112 @@
-                                                                          cred->secret_access_key.c_str());
-        _client = std::make_shared<Aws::S3::S3Client>(credentials, _config);
-    } else {
-        // if not cred provided, we can use default cred in aws profile.
-        _client = std::make_shared<Aws::S3::S3Client>(_config);
+    Status delete_bucket(const std::string& bucket) override;
+
+    StatusOr<std::unique_ptr<io::OutputStream>> put_object(const std::string& bucket,
+                                                           const std::string& object) override;
+
+    StatusOr<std::unique_ptr<io::RandomAccessFile>> get_object(const std::string& bucket,
+                                                               const std::string& object) override;
+
+    Status upload_object(const std::string& bucket, const std::string& object, const std::string& local_path) override;
+
+    Status download_object(const std::string& bucket, const std::string& object,
+                           const std::string& local_path) override;
+
+    Status exist_object(const std::string& bucket, const std::string& object) override;
+
+    StatusOr<int64_t> get_object_size(const std::string& bucket, const std::string& object) override;
+
+    Status delete_object(const std::string& bucket, const std::string& object) override;
+
+    Status list_objects(const std::string& bucket, const std::string& object_prefix,
+                        std::vector<std::string>* result) override;
+
+    StatusOr<std::unique_ptr<io::OutputStream>> put_object(const std::string& uri) override;
+
+    StatusOr<std::unique_ptr<io::RandomAccessFile>> get_object(const std::string& uri) override;
+
+    StatusOr<int64_t> get_object_size(const std::string& uri) override;
+
+    Status exist_object(const std::string& uri) override;
+
+    Status delete_object(const std::string& uri) override;
+
+    Status upload_object(const std::string& uri, const std::string& local_path) override;
+
+    Status download_object(const std::string& uri, const std::string& local_path) override;
+
+private:
+    // transfer manager's thread pool.
+    static const int kThreadPoolNumber = 16;
+    // maximum size of the transfer manager's working buffer to use.
+    static const int kTransferManagerMaxBufferSize = 512 * 1024 * 1024; // 256MB
+    // maximum size that transfer manager will process in a single request.
+    static const int kTransferManagerSingleBufferSize = 32 * 1024 * 1024; // 32MB
+    // return how many keys each time call list_object.
+    static const int kListObjectMaxKeys = 1000;
+
+    static Aws::Utils::Threading::Executor* _get_transfer_manager_executor() {
+        static Aws::Utils::Threading::PooledThreadExecutor executor(kThreadPoolNumber);
+        return &executor;
     }
 
+    Status _head_object(const std::string& bucket, const std::string& object, size_t* size);
```

Comment:
Private method does not need `_` prefix?

Reply:
updated